src/main/java/com/teragrep/rlp_03/Server.java

```diff
@@ -115,6 +115,8 @@ public void run() {
             while (!stop.get()) {
                 socketPoll.poll();
             }
+            // shutdown executorService when outside of poll() loop
+            executorService.shutdown();
```

Comment:
actually when rethinking the finally {} might be even more appropriate so the executorService would be closed even if socketPoll throws